autoload/maktaba/syscall.vim

```diff
@@ -131,8 +131,11 @@ function! maktaba#syscall#DoCallAsync() abort dict
       call s:plugin.logger.Warn('Async support not available. ' .
           \ 'Falling back to synchronous execution for system call: ' .
           \ self.GetCommand())
-      let l:return_data = self.Call()
+      " This is called by DoSyscallCommon, which will throw if v:shell_error is
+      " not 0, so we reset it by executing true.
+      let l:return_data = self.Call(0)  " Don't throw ShellError on failure.
       let l:return_data.status = v:shell_error
+      call maktaba#syscall#Create(['true']).Call()
```

Comment:
This is a little too hacky. Should just call `s:DoSyscallCommon` with `throw_errors=0`. BTW, I have a change in progress to clean up a few corner cases I noticed, so don't sweat the corner cases too much in this change.